tft.py

```diff
@@ -0,0 +1,189 @@
+import json
+import glob
+import mmap
+import copy
+import re
+import os
+from json import JSONEncoder
+from cdragontoolbox.binfile import BinFile, BinHashBase, BinHashValue
+
+
+class NaiveJsonEncoder(JSONEncoder):
+    def default(self, other):  # pylint: disable=method-hidden
+        if isinstance(other, BinHashBase):
+            if other.s is None:
+                return other.hex()
+            return other.s
+        return other.__dict__
+
+
+class TFTTransformer:
+    def parse(self, input):
+        map22_file = os.path.join(input, "data", "maps", "shipping", "map22", "map22.bin")
+        character_folder = os.path.join(input, "data", "characters")
+
+        map22 = BinFile(map22_file)
+        items = self.parse_items(map22)
+        traits = self.parse_traits(map22)
+        champs = self.parse_champs(map22, traits, character_folder)
+
+        # clean up data for JSON export
+        [x.pop("internal") for x in traits]
+        [x.pop("internal") for x in items]
+
+        self.data = {"items": items, "traits": traits, "champs": champs}
+
+    def export(self, output, font_files=[], output_template=False):
```

Comment:
Avoid mutable values are default parameters as they can introduce sneaky bugs if misused.
Default parameters are evaluated once (when the method is parsed), and since lists are used by reference, the same list object will be used for every call.
In your specific case this does not matter since you don't modify `font_files` but it's easy to forget when you change the code so it's considered a good practice to avoid `[]` as default value.

In your case you can simply use `font_files=None` and use `if not font_files: return` before the `for` loop.